src/test/HostActivationTests/NativeHosting/Nethost.cs

```diff
@@ -58,6 +59,38 @@ public void GetHostFxrPath_DotNetRootEnvironment(bool explicitLoad, bool useAsse
             }
         }
 
+        [Theory]
+        [InlineData(true, false, true)]
+        [InlineData(true, false, false)]
+        [InlineData(true, true, true)]
+        [InlineData(true, true, false)]
+        [InlineData(false, false, true)]
+        [InlineData(false, false, false)]
+        [InlineData(false, true, true)]
+        [InlineData(false, true, false)]
+        public void GetHostFxrPath_DotNetRootParameter(bool explicitLoad, bool useAssemblyPath, bool isValid)
```

Comment:
So for my understanding, this handles the case where we provide a path to `dotnet.exe`. I assume this has to be an absolute path, correct? 

I'm still concerned about how path probing works with nethost.dll. If I have a dotnet.exe path that is on `PATH` but isn't anywhere else, will nethost.dll be able to find it?

Reply:
It would have to be absolute or relative (to the current directory). nethost does not look at `PATH` at all (same for the hosting layer in general). You would still need to keep the work that you mentioned exists in ASP.NET to make the prepended `PATH` resolution to a custom location work.